statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,257 @@
+import numpy as np 
+import pandas as pd
+#from numbers import Number
+
+
+class SurveyStat(object):
+
+    def __init__(self, data, cluster=None, strata=None, prob_weights=None):
+
+        data, self.cluster = self._check_type(data, cluster)
+        data, self.strata = self._check_type(data, strata)
+        data, self.prob_weights = self._check_type(data, prob_weights)
+        
+        self.p = data.shape[1]
+        self.data = np.asarray(data)
+
+    # preprocessing to be used by init
+    def _check_type(self, data, vname):
+        """
+        converts cluster, strata, etc to an numpy array
+        gets rid of column from data if possible
+
+        Parameters
+        ----------
+        data : array-like
+          The raw data
+        vname : one of cluster, strata, or prob_weight
+
+        Returns
+        -------
+        A nx1 numpy array
+        """
+        # check if str, num, or list
+
+        self.n = data.shape[0]
+        if isinstance(vname, str):
+            temp = data[vname].values
+            del data[vname]
+            vname = temp
+            n = len(vname)
+            vname = vname.reshape(self.n,1)  
+        elif isinstance(vname, list):
+            vname = np.array(vname).reshape(self.n,1)
+        elif vname is None:
+            vname = np.ones([self.n,1])
+        else:
+            vname = vname.reshape(self.n,1)
+
+        return data, vname
+
+
+    # def create_subgroup_labels(self):
+    #     """
+    #     Create unique integer id for each stratum x cluster combination
+        
+    #     The ids are 0, 1, ... and are kept as an attribute
+    #     """
+
+    #     _, i1 = np.unique(self.strata, return_inverse=True)
```

Comment:
Below is a more complete demonstration of how to use numpy to create unique stratum and cluster indicators efficiently.  We need to be able to handle the situation where the PSU numbers are recycled by strata (e.g. there is a PSU labelled "1" in multiple strata, but they need to be treated as distinct PSUs).  

```
strata = np.r_[1, 1, 1, 1, 1, 3, 3, 3, 3, 3]
cluster = np.r_[1, 1, 2, 2, 2, 1, 1, 2, 2, 2]

_, strata = np.unique(strata, return_inverse=True)
_, cluster = np.unique(cluster, return_inverse=True)

# Recode the PSUs to be sure that the same PSU number in different                                                                                   
# strata are treated as distinct PSUs.                                                                                                               
m = max(cluster)
cluster += (m + 1)*strata
_, cluster = np.unique(cluster, return_inverse=True)
```